helper_test.go

```diff
@@ -1161,6 +1165,74 @@ func createVariableSetVariable(t *testing.T, client *Client, vs *VariableSet, op
 	}
 }
 
+func waitForSVOutputs(t *testing.T, client *Client, svID string) {
+	t.Helper()
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+
+	go retry(func() (interface{}, error) {
+		outputs, err := client.StateVersions.ListOutputs(context.Background(), svID, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(outputs.Items) == 0 {
+			return nil, errors.New("no state version outputs found")
+		}
+
+		wg.Done()
+
+		return outputs, nil
+	})
+
+	wg.Wait()
+}
+
+func waitForRunLock(t *testing.T, client *Client, workspaceID string) {
+	t.Helper()
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+
+	go retry(func() (interface{}, error) {
+		ws, err := client.Workspaces.ReadByID(context.Background(), workspaceID)
+		if err != nil {
+			return nil, err
+		}
+
+		if !ws.Locked {
+			return nil, errors.New("workspace is not locked by run")
+		}
+
+		wg.Done()
+		return ws, nil
+	})
+
+	wg.Wait()
+}
+
+func retry(f retryableFn) (interface{}, error) {
+	tick := time.NewTicker(2 * time.Second)
+
+	retries := 0
+	maxRetries := 5
```

Comment:
How did we determine to set the max retries to 5? Is this value something we'd want to set in an external config file or via an environment variable?

Reply:
We could, however these retries are for select use cases in our test where some action in the API doesn't happen instantaneously. I think waiting worst case ~10 seconds is ample amount of time. If certain async operations are taking longer than that, there might be an underlying issue with the API itself (I think ?).

Reply:
Got it - thanks.